### PR TITLE
add more info about peer discovery

### PR DESCRIPTION
**Pull request by macfarla**
Signed-off-by: Sally MacFarlane <sally.macfarlane@consensys.net>

Added some more info and headings.
Fixed a broken link
Fixes https://github.com/ConsenSys/tessera/issues/1386

### CODE REVIEW

File: docs/Concepts/p2p-discovery.md [hunk 1]
```diff
@@ -4,22 +4,26 @@ description: P2P Tessera Node Discovery
...
+
 A dropped remote peer is added to the `NetworkStore` list only after establishing direct communication with
 the peer. That is, discovering a dropped remote peer is not enough for a node to be added to the active peer list.
 
-The [`partyinfo` API method](https://consensys.github.io/doc.tessera/#operation/getPartyInfo) returns nodes
-with which Tessera has a current active connection (that is, nodes listed in `NetworkStore`).
-
-[Configure peer discovery in the Tessera configuration file](../HowTo/Configure/Peer-discovery.md).
+## Multi-tenancy
+Adding a new key to a multi-tenant Tessera node (and restarting that node) will result in the new key being propagated to other nodes in the network via peer discovery.
```
Comment (alexandratran) [inline docs/Concepts/p2p-discovery.md:29]:
```suggestion

Adding a new key to a multi-tenant Tessera node (and restarting that node) results in the new key being propagated to other nodes in the network via peer discovery.

```

File: docs/Concepts/p2p-discovery.md [hunk 2]
```diff
@@ -4,22 +4,26 @@ description: P2P Tessera Node Discovery
...
 
+Nodes in the network do not all need to have the same list of [configured peers](../HowTo/Configure/Peer-discovery.md#specify-peers). You can think of this list as the starting point for discovering other nodes, similar to bootnodes in [Hyperledger Besu](https://besu.hyperledger.org/en/stable/HowTo/Find-and-Connect/Bootnodes/) or [GoQuorum](https://consensys.net/docs/goquorum/en/stable/configure-and-manage/configure/bootnodes/).
+
+The [`partyinfo` API method](https://consensys.github.io/doc.tessera/#operation/getPartyInfo) returns nodes
+with which Tessera has a current active connection.
+
+[Configure peer discovery](../HowTo/Configure/Peer-discovery.md) in the Tessera configuration file.
+
+## Under the hood
```
Comment (alexandratran) [inline docs/Concepts/p2p-discovery.md:16]:
```suggestion
## Under the hood

```

File: docs/Concepts/p2p-discovery.md [hunk 3]
```diff
@@ -4,22 +4,26 @@ description: P2P Tessera Node Discovery
...
 Tessera maintains two node lists, `PartyStore` and `NetworkStore`. `NetworkStore` lists nodes with
 which an active connection has been established. `PartyStore` lists URLs from the [`peer` entry in the Tessera configuration file](../HowTo/Configure/Peer-discovery.md#specify-peers)
 and URLs discovered from remote nodes. If Tessera can't communicate with a node, the peer is
 removed from both the `PartyStore` and `NetworkStore` lists.
 
 If all peers are removed from the `PartyStore` and `NetworkStore` lists, the `PartyStore` list is repopulated
-from the [`peer` entry in the Tessera configuration file](../HowTo/Configure/Peer-discovery.md#specify-peers).
+from the [`peer` entry in the Tessera configuration file](../HowTo/Configure/Peer-discovery.md#specify-peers).
```
Comment (alexandratran) [inline docs/Concepts/p2p-discovery.md:23]:
```suggestion
from the [`peer` entry in the Tessera configuration file](../HowTo/Configure/Peer-discovery.md#specify-peers).
```

**Review by alexandratran — APPROVED** [review]
Suggestions to fix some CI errors.

File: docs/Concepts/p2p-discovery.md [hunk 4]
```diff
@@ -4,22 +4,26 @@ description: P2P Tessera Node Discovery
 
 # Node discovery
 
-Tessera uses peer discovery to discover Tessera nodes in the network. Tessera nodes share their
-entire peer lists, enabling new nodes to discover all nodes in the network. When sharing their peer lists,
-Tessera broadcasts the public keys for each node and the list of URLs with which each node has
-current active communication.
+Tessera peer discovery works in a similar way to peer discovery in [Hyperledger Besu](https://besu.hyperledger.org/en/stable/HowTo/Find-and-Connect/Managing-Peers/#manage-peers) and [GoQuorum](https://consensys.net/docs/goquorum/en/stable/configure-and-manage/configure/bootnodes/). Tessera nodes share their list of peer URLs, as well as the public keys of those peers. In this way, nodes joining the network are able to discover other nodes in the network, as well as the public keys of other participants.
```
[the remark this answers is not in the document]
Comment (NicolasMassart) [reply docs/Concepts/p2p-discovery.md:7]: Please also split line

**Review by NicolasMassart — COMMENTED** [review]
All test must pass before merge

**Pull request merged**